sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriteFiles.scala

```diff
@@ -38,13 +38,18 @@ case class WriteFilesSpec(
     description: WriteJobDescription,
     committer: FileCommitProtocol,
     concurrentOutputWriterSpecFunc: SparkPlan => Option[ConcurrentOutputWriterSpec])
-  extends WriteSpec
 
 /**
  * During Optimizer, [[V1Writes]] injects the [[WriteFiles]] between [[V1WriteCommand]] and query.
  * [[WriteFiles]] must be the root plan as the child of [[V1WriteCommand]].
  */
-case class WriteFiles(child: LogicalPlan) extends UnaryNode {
+case class WriteFiles(
+    child: LogicalPlan,
+    fileFormat: FileFormat,
+    partitionColumns: Seq[Attribute],
+    bucketSpec: Option[BucketSpec],
+    options: Map[String, String],
+    requiredOrdering: Seq[SortOrder]) extends UnaryNode {
```

Comment:
This doesn't seem like a logical write information, but more of internal information. Do we really need it here?

Reply:
how about pull out `partitionSpec` instead ? `partitionColumns` does not contain the information of the insertion partition spec.

Reply:
SGTM